src/main/java/com/adyen/model/checkout/PaymentMethodDetails.java

```diff
@@ -21,6 +21,5 @@
 package com.adyen.model.checkout;
 
 public interface PaymentMethodDetails {
```

Comment:
I do not see the point of an empty interface just for the sake of convenience. I suggest the following:

- Make it a generic interface to support the different types of the property "type", so we can still keep the required methods to be implemented.
- Add comments that give a hint to the minimum required methods to be created by a class implementing this interface.

Otherwise this interface does not have much of a purpose, from an interface point of view.

Reply:
can't we add the type as a default get/setter to implement ?

Reply:
some classes have enums as type, so we cannot use String in the interface

Reply:
Yeah we can add empty default methods as a hint, but then when implementing the interface you would have to override your getters/setters as to not use the default ones, which is a bit counter-intuitive in my opinion.